plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -480,10 +487,40 @@ private Optional<Table> loadTable(HiveTableName hiveTableName)
         return delegate.getTable(hiveTableName.getDatabaseName(), hiveTableName.getTableName());
     }
 
+    /**
+     * The method will cache and return columns specified in the {@link Table#getDataColumns()}
+     * but may return more if other columns are already cached.
+     */
     @Override
     public PartitionStatistics getTableStatistics(Table table)
     {
-        return get(tableStatisticsCache, hiveTableName(table.getDatabaseName(), table.getTableName()), () -> delegate.getTableStatistics(table));
+        HiveTableName tableName = hiveTableName(table.getDatabaseName(), table.getTableName());
+        PartitionStatistics statistics = get(tableStatisticsCache, tableName, () -> delegate.getTableStatistics(table));
+        Set<String> dataColumns = table.getDataColumns().stream().map(Column::getName).collect(toImmutableSet());
+        if (statistics.getColumnStatistics().keySet().containsAll(dataColumns)) {
```

Comment:
yes, that was intentional to avoid copying the stats in the steady-state case where most of the columns are cached.
Do you think its a big issue?

Reply:
I was thinking if it would be beneficial for rest of trino to not return 100 columns statistics instead of 1 (yours benchmark example).

Could you test what is the difference between set of queries with 1 column and all columns vs all columns, 1 column?

Reply:
> I was thinking if it would be beneficial for rest of trino to not return 100 columns statistics instead of 1 (yours benchmark example).

The 100 columns are there anyway, I return a reference. What is the benefit of returning just a limited set of columns?

> Could you test what is the difference between set of queries with 1 column and all columns vs all columns, 1 column?

Do you mean to compare query latency? It's highly unlikely this will have a visible impact either way

Reply:
First set will trigger getting additional columns and merge them with existing cache, second set will present old behavior (first query will fill cache for all columns) and second query will run from cache. I want to see how beneficial this mechanism is.